climada_petals/hazard/coastal_flood.py

```diff
@@ -0,0 +1,127 @@
+from typing import Iterable, Union, Optional
+import numpy as np
+from shapely.geometry import Polygon
+
+from climada.hazard import Hazard
+from climada.util import files_handler as u_fh
+import climada.util.coordinates as u_coord
+from climada import CONFIG
+
+AQUEDUCT_SOURCE_LINK = CONFIG.hazard.coastal_flood.resources.aqueduct.str()
+DOWNLOAD_DIRECTORY = CONFIG.hazard.coastal_flood.local_data.aqueduct.dir()
+
+HAZ_TYPE = 'CF'
+"""Hazard type acronym CoastalFlood"""
+
+__all__ = ['CoastalFlood']
+
+class CoastalFlood(Hazard):
+    """
+    Contains coastal flood events pulled by the Acqueduct project :
+        https://www.wri.org/data/aqueduct-floods-hazard-maps
+    """
+
+    def __init__(self, **kwargs):
+        """Empty constructor"""
+
+        Hazard.__init__(self, HAZ_TYPE, **kwargs)
+
+    @classmethod
+    def from_aqueduct_tif(cls,
```

Comment:
In this case I think it makes much more sense to remove the `from` method and put all the functionality in the init. There seems to be no other purpose than to load data (initialize the object).

Reply:
mhh, as of now maybe yes, but it's likely that the CoastalFlood class will be populated with more data in the future; same as it's happening with the RiverFlood class where we now have ISIMIP and Aqueduct (https://github.com/CLIMADA-project/climada_petals/blob/feature/river_flooding_aqueduct/climada_petals/hazard/river_flood.py) so I would tend to keep it like this.